src/property.cxx

```diff
@@ -256,27 +256,53 @@ Properties::init(tistream& input)
             trim_ws (value);
             setProperty(key, value);
         }
-        else if (buffer.compare (0, 7, LOG4CPLUS_TEXT ("include")) == 0
-            && buffer.size () >= 7 + 1 + 1
-            && is_space (buffer[7]))
-        {
-            tstring included (buffer, 8) ;
-            trim_ws (included);
+		else if (buffer.compare(0, 7, LOG4CPLUS_TEXT("include")) == 0
+			&& buffer.size() >= 7 + 1 + 1
+			&& is_space(buffer[7]))
+		{
+			tstring included(buffer, 8);
+			trim_ws(included);
+
+
+			tstring fileName = included;
+			tstring modFilename;
+			while (substitueEnvVariables(fileName, modFilename))
+			{
+				fileName = modFilename;
+			}
 
             tifstream file;
             imbue_file_from_flags (file, flags);
 
-            file.open (LOG4CPLUS_FSTREAM_PREFERED_FILE_NAME(included).c_str(),
+            file.open (LOG4CPLUS_FSTREAM_PREFERED_FILE_NAME(fileName).c_str(),
                 std::ios::binary);
             if (! file.good ())
                 helpers::getLogLog ().error (
-                    LOG4CPLUS_TEXT ("could not open file ") + included);
+                    LOG4CPLUS_TEXT ("could not open file ") + fileName);
 
             init (file);
         }
     }
 }
 
+bool
+Properties::substitueEnvVariables(tstring &fileName, tstring &modFileName)
```

Comment:
I am not sure that we need a new function for this. There is already a function [`substVars()`](https://github.com/log4cplus/log4cplus/blob/078ee67544151020528458f3ef9185d2fa632797/src/configurator.cxx#L102) in `configurator.cxx`. It seems it should be used here as well. If so, you can probably move it from anonymous namespace into `internal` namespace and make it visible to the `property.cxx` and use it there.